lib/spack/spack/directives.py

```diff
@@ -180,24 +180,43 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and link because the common case is to
+        # build against a library which then turns into a runtime dependency
+        # due to the linker.
```

Comment:
looks like an outdated comment. From the description of `run` and `link`, i got an impression that `run` is primarily for extensions of python and alike.

i suppose `mpi` would be `build`+`link`+`run`?